### DISCUSSION TITLE
Coupling MOOSE and PHREEQC for reservoir reactive fluid flow.

### DISCUSSION THREAD

**Discussion by Keshi-23** [idaholab/moose#26109] [Q&A Modules: Porous Flow]
Hi,
I would like to understand and get expert guidance on how I can couple the MOOSE PorousFlow module with phreeqc for modeling reactive fluid flow.

@GiudGiud @cpgr

Thanks

**Comment by cpgr** [comment]
I've never tried this, and have never used phreeqc. Can you give me an idea of what you are trying to do?

**Comment by josebastiase** [comment]
I don't think that is possible right now, but I know some people at the [GFZ](https://www.gfz-potsdam.de/en/) who are working exactly on this.

It seems like you need to output variables from both codes after each time step and update the boundary conditions to have sequential loose coupling between the codes (a co-simulation). This is possible in MOOSE with a Userobject; just add it to your PorousFlow app. I have done it with other codes.

An option for COMSOL is [CPqPy](https://www.sciencedirect.com/science/article/pii/S0098300414000880).

If you know C++, you can also fully couple the codes by linking both libraries together, but this is definitely harder.

Finally, it's probably better if you use Geochemistry. As native MOOSE code, it will likely be faster, and the MOOSE team has put a lot of effort into making MultyApps smooth and very functional. You can definitely benefit from that capacity too.

I hope it helps.
